### ISSUE TITLE
New Unicode channel names are not supported and lead to disconnection

### ISSUE THREAD

**Issue by thor** [wee-slack/wee-slack#324]
Slack [recently enabled users to go full Unicode in their channel names](https://twitter.com/SlackHQ/status/842475876134481920).

This has had the unfortunate consequence that wee-slack seems to go into a state of disconnection once it has enumerated through the list of of channels and reached one with unicode in it.

It also provides the following, not surprising, traceback if you try to join a channel with Unicode in it:
```
08:07 python: stdout/stderr: __weechat_plugin__:445: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
08:07 python: stdout/stderr: __weechat_plugin__:899: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
08:07 python: stdout/stderr: Traceback (most recent call last):
08:07 python: stdout/stderr:   File "/home/thor/.weechat/python/autoload/slack_extension.py", line 1049, in wrapper
08:07 python: stdout/stderr:     return f(current_buffer, *args, **kwargs)
08:07 python: stdout/stderr:   File "/home/thor/.weechat/python/autoload/slack_extension.py", line 1083, in join_command_cb
08:07 python: stdout/stderr:     elif command_talk(current_buffer, args[1]):
08:07 python: stdout/stderr:   File "/home/thor/.weechat/python/autoload/slack_extension.py", line 1198, in command_talk
08:07 python: stdout/stderr:     server.buffer_prnt("User or channel {} not found.".format(args))
08:07 python: stdout/stderr:   File "/home/thor/.weechat/python/autoload/slack_extension.py", line 364, in buffer_prnt
08:07 python: stdout/stderr:     message = message.encode('ascii', 'ignore')
08:07 python: stdout/stderr: UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 17: ordinal not in range(128)
08:07 =!= python: error in function "join_command_cb"
```

**Comment by trygveaa** [comment]
I think this is fixed with #348.

**Comment by thor** [comment]
@trygveaa I think you're quite likely right! I at the very least cannot reproduce this, as unicode channel names works great with the current codebase. 👍 Closing the issue.
